src/docfx/lib/JsonUtility.cs

```diff
@@ -205,8 +208,58 @@ protected override JsonProperty CreateProperty(MemberInfo member, MemberSerializ
                         prop.Writable = true;
                     }
                 }
+
+                if (converter != null)
+                {
+                    if (prop.PropertyType.IsArray)
+                    {
+                        prop.ItemConverter = converter;
+                    }
+                    else
+                    {
+                        prop.Converter = converter;
+                    }
+                }
                 return prop;
             }
+
+            private static SchemaValidationConverter GetConverter(MemberInfo member)
+            {
+                var validators = member.GetCustomAttributes<ValidationAttribute>();
+                return !validators.Any() ? null : new SchemaValidationConverter(validators);
```

Comment:
>validators [](start = 24, length = 10)

validators.Length == 0

Reply:
The type of validators is IEnumerable<T>, Length is not available here, Any() would just iterate once with the first element

---
In reply to: [206377491](https://github.com/dotnet/docfx/pull/3130#discussion_r206377491) [](ancestors = 206377491)

Reply:
can `ToList()` first. Using IEnumerable twice will call `member.GetCustomAttributes<ValidationAttribute>(false);` twice.

---
In reply to: [206379098](https://github.com/dotnet/docfx/pull/3130#discussion_r206379098) [](ancestors = 206379098,206377491)

Reply:
I suppose GetCustomAttributes should have yield return inside, since the return type is IEnumerable<T>, .Any() will add one more iteration, not the entire collection :)  Anyway, ToList will  avoid this additional iteration, I will add it.

---
In reply to: [206410955](https://github.com/dotnet/docfx/pull/3130#discussion_r206410955) [](ancestors = 206410955,206379098,206377491)

Reply:
seems no need to `ToList()`?
[https://docs.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.getcustomattributes?view=netframework-4.7.2#System_Reflection_MemberInfo_GetCustomAttributes_System_Boolean](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.getcustomattributes?view=netframework-4.7.2#System_Reflection_MemberInfo_GetCustomAttributes_System_Boolean)_

---
In reply to: [206421770](https://github.com/dotnet/docfx/pull/3130#discussion_r206421770) [](ancestors = 206421770,206410955,206379098,206377491)

Reply:
We want to use the extension method with generic type here, because we want to use ValidationAttribute afterwards without casting.
[https://msdn.microsoft.com/en-us/library/hh138362(v=vs.110).aspx](https://msdn.microsoft.com/en-us/library/hh138362(v=vs.110).aspx)
Basically with ToList(), we can have Count in place without iteration, but we created a new list. For IEnumerable<T>, while invoke .Any(), we will need 1 additional iteration to check if any element inside.
[https://stackoverflow.com/questions/305092/which-method-performs-better-any-vs-count-0](https://stackoverflow.com/questions/305092/which-method-performs-better-any-vs-count-0)

---
In reply to: [206512597](https://github.com/dotnet/docfx/pull/3130#discussion_r206512597) [](ancestors = 206512597,206421770,206410955,206379098,206377491)

Reply:
sure, in this case, I also prefer use Any(). thanks for detailed explanation!

---
In reply to: [206736462](https://github.com/dotnet/docfx/pull/3130#discussion_r206736462) [](ancestors = 206736462,206512597,206421770,206410955,206379098,206377491)